rosbag2_storage_sqlite3/src/rosbag2_storage_sqlite3/sqlite_storage.cpp

```diff
@@ -852,6 +878,36 @@ int SqliteStorage::read_db_schema_version()
   return schema_version;
 }
 
+uint64_t SqliteStorage::read_total_page_count_locked()
+{
+  auto scope_exit_reset_statement = rcpputils::make_scope_exit(
+    [this]() {
+      page_count_statement_->reset();
+    });
```

Comment:
Why are we resetting `page_count_statement` after we use it?  I don't know this code very well, but it seems like this statement should just live for the lifetime of the object.

Reply:
Yeah. I also was confused about this reset when was started implementing this feature and was trying to keep the statement state between invocations. Then I've got some intermittent errors with a message that the database is locked and can't perform other operations on it.  
After some investigation, it turns out that this reset is required for each statement invocation and part of the SQLite API workflow.  
From the `sqlite3.h`
> ** The sqlite3_reset() function is called to reset a [prepared statement]
** object back to its initial state, ready to be re-executed.
** ^Any SQL statement variables that had values bound to them using
** the [sqlite3_bind_blob | sqlite3_bind_*() API] retain their values.
** Use [sqlite3_clear_bindings()] to reset the bindings.
**
** ^The [sqlite3_reset(S)] interface resets the [prepared statement] S
** back to the beginning of its program.
**
** ^If the most recent call to [sqlite3_step(S)] for the
** [prepared statement] S returned [SQLITE_ROW] or [SQLITE_DONE],
** or if [sqlite3_step(S)] has never before been called on S,
** then [sqlite3_reset(S)] returns [SQLITE_OK].
**
** ^If the most recent call to [sqlite3_step(S)] for the
** [prepared statement] S indicated an error, then
** [sqlite3_reset(S)] returns an appropriate [error code].